File: Retrofit/retrofitex/ApiClient.java
package com.example.retrofitex;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient
{
    public static final String BASE_URL="https://vyasprakruti.000webhostapp.com/30Nov/";
    public static Retrofit retrofit=null;

    public static Retrofit getapiclient()
    {
        if(retrofit==null)
        {
            retrofit= new Retrofit.Builder().
                    baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();

        }


        return retrofit;
    }
}

File: 31-8/Project/main/java/com/example/e_greetings/Other/ApiClient.kt
package com.example.e_greetings.Other

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class ApiClient
{

    companion object
    {

        var url="https://vyasprakruti.000webhostapp.com/E-Greetings%20Project/"
        fun getapiclient(): Retrofit
        {
            var retrofit:Retrofit?=null

            if(retrofit==null)
            {
                retrofit=Retrofit.Builder()
                    .baseUrl(url)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
            }


            return retrofit!!
        }
    }



}